src/cart/crt_hg_proc.c

```diff
@@ -541,6 +546,8 @@ crt_hg_unpack_header(hg_handle_t handle, struct crt_rpc_priv *rpc_priv,
 		D_ERROR("crt_proc_common_hdr failed rc: %d.\n", rc);
 		D_GOTO(out, rc);
 	}
+	rpc_priv->crp_reply_hdr.cch_hlc = crt_hlc_get_msg(
+						rpc_priv->crp_req_hdr.cch_hlc);
```

Comment:
so this member is being reused to store the received HLC, so later call of crt_hlc_receive_get() can extract it from this message right?

The HLC of the outgoing reply will be overwritten while packing the reply?

Reply:
I think here it is just to update local crt_htc, the assignment to "crp_reply_hdr.cch_hlc" seems useless here, as later when sending the reply in crt_proc_out_common it will re-assign it: crp_reply_hdr.cch_hlc = crt_hlc_get();

Reply:
Here I update a local HLC and store receiving timestamp which will be get by crt_hlc_receive_get(rpc). This is receive new RPC code path. For replys it will be set in different place (@liuxuezhao point to this case). But this is not overriding. This is two different code paths.

Reply:
Hmm. I need to refine this.

Reply:
reworked this.